### PR TITLE
Add material-ui themed search

### PR DESCRIPTION
**Pull request by peatroot**
This PR:
* adds a Search component that builds on react-select's AsyncSelect component
* adds subcomponents with material-ui styling as described [here](https://material-ui.com/demos/autocomplete/#react-select)
* exposes a minimal set of props to the using application (optionComponent, onInputChange, onSelectOption, value, placeholder)

### CODE REVIEW

File: lib/components/search/Search.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    } = this.props;
+
+    const selectStyles = {
+      input: base => ({
+        ...base,
+        color: theme.palette.text.primary,
+      }),
+    };
+
+    const Option = props => (
```
Comment (mirandaio) [inline lib/components/search/Search.js]: This component could be declared outside the component. What do you think?

File: lib/components/search/Search.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+const styles = theme => ({
+  root: {
+    position: 'relative',
+  },
+});
+
+const IndicatorSeparator = () => null;
+
+class Search extends React.Component {
```
Comment (mirandaio) [inline lib/components/search/Search.js]: The `Search` component can be a functional component.

File: lib/components/search/Control.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+};
+
+class InputComponent extends React.Component {
+  render() {
+    const { inputRef, ...rest } = this.props;
+    return <div ref={inputRef} {...rest} />;
+  }
+}
+
+class Control extends React.Component {
```
[the remark this answers is not in the document]
Comment (peatroot) [reply lib/components/search/Control.js]: Actually, there's a funny issue here. If you make Control a functional component, you get console errors of the form:
```
Stateless function components cannot be given refs. Attempts to access this ref will fail.
```
See https://github.com/facebook/react/issues/10831#issuecomment-332824850.

[the remark this answers is not in the document]
Comment (peatroot) [reply lib/components/search/Control.js]: Seems to be fine with InputComponent though

File: lib/components/search/Menu.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+const styles = theme => ({
+  paper: {
+    marginTop: theme.spacing.unit,
+    position: 'absolute',
+    width: '100%',
+    zIndex: 1005,
+  },
+});
+
+const Menu = ({ classes, selectProps, innerProps, children }) => (
```
[the remark this answers is not in the document]
Comment (peatroot) [reply lib/components/search/Menu.js]: 👍

File: lib/components/search/Search.js (hunk 1 above)
Comment (peatroot) [reply lib/components/search/Search.js]: Cannot do this because `Option` depends on the `optionComponent` prop.

Comment (mirandaio) [reply lib/components/search/Search.js]: Ah, overlooked that.

**Pull request merged**